OptimoveSDK/Sources/Classes/PreferenceCenter/OptimovePreferenceCenter.swift

```diff
@@ -0,0 +1,192 @@
+
+    public static func getInstance() throws -> OptimovePreferenceCenter {
+        guard let instance = instance else {
+            throw Error.notInitialized
+        }
+        return instance
+    }
+
+    static func initialize(storage: OptimoveStorage, networkClient: NetworkClient) throws {
+        guard instance == nil else {
+            throw Error.alreadyInitialized
+        }
+
+        instance = OptimovePreferenceCenter(storage: storage, networkClient: networkClient)
+    }
+
+    private init(storage: OptimoveStorage, networkClient: NetworkClient) {
+        self.networkClient = networkClient
+        self.storage = storage
+
+        Logger.debug("Preference center SDK was initialized")
+    }
+
+    public func getPreferencesAsync(completion: @escaping PreferencesGetHandler) {
+        guard let config = Optimove.getConfig()?.getPreferenceCenterConfig() else {
+            Logger.error("Preference center credentials are not set")
+            completion(.errorNotConfigured, nil)
+            return
+        }
+
+        guard
+            let customerId = try? storage?.getCustomerID(),
+            let visitorId = try? storage?.getVisitorID(),
+            customerId != visitorId
+        else {
+            Logger.warn("Customer ID is not set")
+            completion(.errorUserNotSet, nil)
+            return
+        }
+
+        Task {
+            do {
+                let request = createGetPreferencesRequest(for: customerId, with: config)
+                let data = try await networkClient?.performAsync(request) ?? Data()
+                let preferences = try JSONDecoder().decode(Preferences.self, from: data)
+
+                DispatchQueue.main.async {
+                    completion(.success, preferences)
+                }
+            } catch {
+                DispatchQueue.main.async {
+                    completion(.error, nil)
+                }
+            }
+        }
+    }
+
+    private func createGetPreferencesRequest(for customerId: String, with config: PreferenceCenterConfig) -> NetworkRequest {
+        do {
+            let (region, brandGroupId, tenantId) = try getConfigValues(from: config)
+
+            return NetworkRequest(
+                method: .get,
+                baseURL: URL(string: "https://preference-center-\(region).optimove.net")!,
+                path: "/api/v1/preferences",
+                headers: [
+                    HTTPHeader(field: .accept, value: .textplain),
+                    HTTPHeader(field: .tenantId, value: .tenantId(id: tenantId))
+                ],
+                queryItems: [
+                    URLQueryItem(name: "customerId", value: customerId),
+                    URLQueryItem(name: "brandGroupId", value: brandGroupId)
+                ]
+            )
+        } catch {
+            fatalError("Configuration is missing. This should never happen if the SDK is initialized properly.")
```

Comment:
does this one crash the app? If we go consistently about "PC never crashes the app" -- shall just log